optimum/habana/transformers/models/mamba/modeling_mamba.py

```diff
@@ -1,16 +1,76 @@
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
 
 import torch
-from transformers.models.mamba.modeling_mamba import (
-    MambaCache,
-)
+from torch import nn
+from transformers.cache_utils import MambaCache
+from transformers.configuration_utils import PretrainedConfig
+from transformers.models.mamba.configuration_mamba import MambaConfig
+from transformers.activations import ACT2FN
 from transformers.utils import (
     ModelOutput,
     logging,
 )
 
+from pathlib import Path
+import os
+base_dir = "/workspace/custom_op_pscan_all" 
```

Comment:
@libinta what would be the best way to set this without hardcoding?
Atleast an env var?

Reply:
Is this dir generated on the fly? Or is it supposed to be downloaded (e.g. as part of an example) ?

Reply:
We can change to
my_dir = os.path.realpath(__file__)
my_len = my_dir.rfind("/")
base_dir = my_dir[:my_len]